tests/gtest/avifgainmaptest.cc

```diff
@@ -351,5 +354,113 @@ TEST(GainMapTest, SequenceNotSupported) {
       << avifResultToString(result) << " " << encoder->diag.error;
 }
 
+TEST(GainMapTest, IgnoreGainMapOrColor) {
```

Comment:
Just wondering if that test and the following could be merged as a parametrized test depending on:
ignoreGainMap, ignoreColorAndAlpha and the creation of a gain map.
Not sure if it'd be easier to read and smaller in size.